compose/snippets/src/main/java/com/example/compose/snippets/topcomponents/AppBar.kt

```diff
@@ -0,0 +1,327 @@
+import androidx.compose.material3.BottomAppBar
+import androidx.compose.material3.BottomAppBarDefaults
+import androidx.compose.material3.Button
+import androidx.compose.material3.CenterAlignedTopAppBar
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.FloatingActionButton
+import androidx.compose.material3.FloatingActionButtonDefaults
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.LargeTopAppBar
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.MediumTopAppBar
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.material3.TopAppBar
+import androidx.compose.material3.TopAppBarDefaults
+import androidx.compose.material3.rememberTopAppBarState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.input.nestedscroll.nestedScroll
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.dp
+
+@Composable
+fun AppBarExamples() {
+    var selection by remember { mutableStateOf("none") }
+
+    Surface(
+        modifier = Modifier.fillMaxSize(),
+        color = MaterialTheme.colorScheme.background
+    ) {
+        when (selection) {
+            "bottomBar" -> BottomAppBarExample()
+            "topBar" -> SmallTopAppBarExample()
+            "topBarCenter" -> CenterAlignedTopAppBarExample()
+            "topBarMedium" -> MediumTopAppBarExample()
+            "topBarLarge" -> LargeTopAppBarExample()
+            else -> AppBarOptions(
+                toBottom = { selection = "bottomBar" },
+                toTopBarSmall = { selection = "topBar" },
+                toTopBarCenter = { selection = "topBarCenter" },
+                toTopBarMedium = { selection = "topBarMedium" },
+                toTopBarLarge = { selection = "topBarLarge" },
+            )
+        }
+    }
+}
+
+@Composable
+fun AppBarOptions(
+    toBottom: () -> Unit,
+    toTopBarSmall: () -> Unit,
+    toTopBarCenter: () -> Unit,
+    toTopBarMedium: () -> Unit,
+    toTopBarLarge: () -> Unit,
+) {
+    Column() {
+        Button({ toBottom() }) {
+            Text("Bottom bar")
+        }
+        Button({ toTopBarSmall() }) {
+            Text("Small top bar")
+        }
+        Button({ toTopBarCenter() }) {
+            Text("Center aligned top bar")
+        }
+        Button({ toTopBarMedium() }) {
+            Text("Medium top bar")
+        }
+        Button({ toTopBarLarge() }) {
+            Text("Large top bar")
+        }
+    }
+}
+
+@Composable
+fun BottomAppBarExample() {
+    Scaffold(
+        bottomBar = {
+            BottomAppBar(
```

Comment:
For each of these snippets, you will need to surround the code with start and end tags for DAC includecode snippet import to work. 
```
// [START android_compose_components_bottombar]
 the code you want displayed on DAC
// [END android_compose_components_bottombar]
```

Reply:
Done.